raft/src/test/java/org/apache/kafka/raft/internals/KafkaRaftMetricsTest.java

```diff
@@ -469,6 +478,54 @@ public void shouldRecordRate(KRaftVersion kraftVersion) {
         assertEquals(2.4, getMetric(metrics, "fetch-records-rate").metricValue());
     }
 
+    @ParameterizedTest
+    @EnumSource(value = KRaftVersion.class)
+    public void testNumberOfVoters(KRaftVersion kraftVersion) {
+        raftMetrics = new KafkaRaftMetrics(metrics, "raft");
+        QuorumState state = buildQuorumState(localStandaloneVoterSet(kraftVersion), kraftVersion);
+        raftMetrics.initialize(state);
+        state.initialize(new OffsetAndEpoch(0L, 0));
+
+        assertEquals(0, getMetric(metrics, "number-of-voters").metricValue());
+
+        raftMetrics.updateNumVoters(3);
+        assertEquals(3, getMetric(metrics, "number-of-voters").metricValue());
+    }
+
+    @ParameterizedTest
+    @EnumSource(value = KRaftVersion.class)
+    public void testLeaderMetrics(KRaftVersion kraftVersion) {
+        raftMetrics = new KafkaRaftMetrics(metrics, "raft");
+        QuorumState state = buildQuorumState(localStandaloneVoterSet(kraftVersion), kraftVersion);
+        raftMetrics.initialize(state);
+        state.initialize(new OffsetAndEpoch(0L, 0));
+
+        assertNull(getMetric(metrics, "number-of-observers"));
+        assertNull(getMetric(metrics, "uncommitted-voter-change"));
+
+        raftMetrics.addLeaderMetrics();
+
+        assertEquals(0, getMetric(metrics, "number-of-observers").metricValue());
+        assertEquals(false, getMetric(metrics, "uncommitted-voter-change").metricValue());
```

Comment:
I made a similar comment in the `src/main` implementation but this metric should be an integer with the values 1 or 0.